frontend/src/pages/categories/Categories.js

```diff
@@ -1,24 +1,33 @@
-import { Card, Sidebar } from "components"
-import { useProduct } from "context/productContext"
-import './categories.css'
+import { getProducts } from "actions/productActions";
+import { Card, Sidebar } from "components";
+import { useProduct } from "context/productContext";
+import { useEffect } from "react";
+import "./categories.css";
 
 export const Categories = () => {
-  const {products} = useProduct()
-    return(
-        <>
-        <main className="container">
+  const { state, dispatch } = useProduct();
+
+  useEffect(() => {
+    getProducts(dispatch);
+  }, []);
+  const { loading, products, error } = state;
+  return (
+    <>
+      <main className="container">
         <Sidebar />
         <div className="main-content">
           {products.map((product, index) => (
```

Comment:
you can de-structure the product object. This way repeatition of product can be avoided